src/test/lambda/local/debugConfiguration.test.ts

```diff
@@ -215,3 +216,111 @@ describe('isImageLambdaConfig', function () {
         assert.strictEqual(isImageLambdaConfig(input), false)
     })
 })
+
+describe('getArchitecture', function () {
+    it('returns a valid architecture from a template', function () {
+        const resource: CloudFormation.Resource = {
+            Type: 'AWS::Serverless::Function',
+            Properties: {
+                CodeUri: 'foo',
+                Handler: 'foo',
+                Architectures: ['arm64'],
+            },
+        }
+        const template: CloudFormation.Template = {
+            Resources: {
+                myResource: resource,
+            },
+        }
+        assert.strictEqual(
+            getArchitecture(template, resource, {
+                target: 'template',
+                logicalId: 'myResource',
+                templatePath: 'foo',
+            }),
+            'arm64'
+        )
+    })
+
+    it('returns x86_64 for an invalid architecture from a template', function () {
+        const resource: CloudFormation.Resource = {
+            Type: 'AWS::Serverless::Function',
+            Properties: {
+                CodeUri: 'foo',
+                Handler: 'foo',
+                Architectures: ['powerPc' as 'x86_64'],
+            },
+        }
+        const template: CloudFormation.Template = {
+            Resources: {
+                myResource: resource,
+            },
+        }
+        assert.strictEqual(
+            getArchitecture(template, resource, {
+                target: 'template',
+                logicalId: 'myResource',
+                templatePath: 'foo',
+            }),
+            'x86_64'
+        )
+    })
+
+    it('returns a valid architecture from CodeTargetProperties', function () {
+        assert.strictEqual(
+            getArchitecture(undefined, undefined, {
+                lambdaHandler: 'foo',
+                projectRoot: 'foo',
+                target: 'code',
+                architecture: 'arm64',
+            }),
+            'arm64'
+        )
+    })
+
+    it('returns x86_64 for an invalid architecture from CodeTargetProperties', function () {
+        assert.strictEqual(
+            getArchitecture(undefined, undefined, {
+                lambdaHandler: 'foo',
+                projectRoot: 'foo',
+                target: 'code',
+                architecture: 'powerPc' as 'x86_64',
```

Comment:
IMO this test should be moved 'up' closer to where the data is coming from. That is, we should validate the user is giving us good data sooner. Then there's no need to hard-cast to test since the types themselves are the test.

Reply:
After looking at the code more this is pretty close to the source. I think the test format was just throwing me off.